service/src/main/java/org/cbioportal/genome_nexus/service/cached/BaseCachedExternalResourceFetcher.java

```diff
@@ -67,6 +70,22 @@ protected Boolean isValidId(String id)
 
     public T fetchAndCache(String id) throws ResourceMappingException
     {
+
+        // If caching is disabled, always fetch directly from the web service
+         if (!cacheEnabled) {
+            try {
+                DBObject rawValue = this.fetcher.fetchRawValue(id);
+                rawValue = this.normalizeResponse(rawValue);
+                List<T> list = this.transformer.transform(rawValue, this.type);
+                if (!list.isEmpty()) {
+                    return list.get(0);
+                }
+            } catch (Exception e) {
+                LOG.error("Error fetching external resource for id " + id, e);
+            }
```

Comment:
This is pretty similar to the code segment below.

https://github.com/genome-nexus/genome-nexus/pull/775/files#diff-497a870f81e19d2415bfdcbd0eb906a6894575c15bdd26b8a5fae22bab86dd9bR107-R133

Might be better to add a new method and reuse in both places.

Reply:
Moved to a new method